### PR TITLE
E2E Tests: Improve flakey Publishing Flow test

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Do not use `Promise.all`. 
> Pass all of them to Promise.all() as an array and run them simultaneously, and process the data once all promises are resolved.

We do not wish `waitForNavigation` to run simultaneously to the togo. The test should await the goto before running `waitForNavigation`. 

Also add `networkidle0` to all goto, to make sure all assets are loaded.

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

**Comment by spacedmonkey** [comment]
I believe that is block by #6238

**Comment by miina** [comment]
@spacedmonkey Is this for the e2e tests that were sometimes failing before? If that's the case, do the tests always pass now or still flaky but less flaky?

File: packages/e2e-tests/src/specs/editor/publishingFlow.js [hunk 1]
```diff
@@ -110,7 +110,9 @@ describe('Publishing Flow', () => {
     expect(postPermalink).not.toBeNull();
     expect(postPermalink).toStrictEqual(expect.any(String));
 
-    await Promise.all([page.goto(postPermalink), page.waitForNavigation()]);
+    await page.goto(postPermalink, {
+      waitUntil: 'networkidle0',
+    });
```
Comment (miina) [inline packages/e2e-tests/src/specs/editor/publishingFlow.js:119]: Would the case be similar here: https://github.com/google/web-stories-wp/blob/a640f7e734ec377c79d7295846d839991d0d5887/packages/e2e-tests/src/specs/wordpress/quickEdit.js#L77-L80

That it should first wait for the click and only then `waitForNavigation()`?

**Comment by spacedmonkey** [comment]
> @spacedmonkey Is this for the e2e tests that were sometimes failing before? If that's the case, do the tests always pass now or still flaky but less flaky?

I have run the test 3 times with no fails. Honestly this is a weird with puppetter and think this fixes it.

File: packages/e2e-tests/src/specs/editor/publishingFlow.js (hunk 1 above)
Comment (swissspidy) [reply packages/e2e-tests/src/specs/editor/publishingFlow.js:119]: @spacedmonkey Could you take a look at all instances of these cases? Would be good to have uniform handling of page navigation in e2e tests.

Comment (spacedmonkey) [reply packages/e2e-tests/src/specs/editor/publishingFlow.js:119]: I did. The issue with this test is, `page.waitForNavigation()` when you click / interact with an element and the browser changes url, but just does do anything with a togo. In this PR, I replace all the places that do this, with a wait until network connections have stopped for 500ms.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**